frontend/js/src/stats/UserReports.tsx

```diff
@@ -97,7 +99,7 @@ export default class UserReports extends React.Component<
   };
 
   render() {
-    const { range, user } = this.state;
+    const { range, user, selectedUser } = this.state;
     const { apiUrl } = this.props;
```

Comment:
I do not think we need this new state variable.
Instead, the issue we have is that we are using the `user` state (which we modify when clicking the buttons) instead of `user` from the props, which does not change.

```suggestion
    const { range, user } = this.state;
    const { apiUrl, user: initialUser } = this.props;
```

With this, `initialUser` is the same as what you wanted to do with the `selectedUser` state and you can use it as you did below for the button.

The `user: initialUser` syntax is to rename the variable when destructuring `this.props.user` so that the variable name does not collide with `this.state.user` declared just above.
For reference, see the "Assigning to new variable names" section in https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment